test/database_service/test_db_experiment_data.py

```diff
@@ -752,7 +751,7 @@ def _job_result():
         exp_data.add_jobs(job, timeout=0.5)
 
         with self.assertLogs("qiskit_experiments", "WARNING"):
-            exp_data.block_for_results()
+            self.assertFail(exp_data)
```

Comment:
This isn't necessary and doesn't improve the readability of these tests. Since the status is already checked in these tests with more detail than the assertFail test. All you are doing here is using it as an opaque way to block for results.